accepted/2023/simplify-workload-versioning.md

```diff
@@ -0,0 +1,153 @@
+
+```
+> dotnet --version
+8.0.201
+```
+
+We will add a new `dotnet workload --version` command to print the workloads version:
+
+```
+> dotnet workload --version
+8.0.201
+```
+
+We will also update `dotnet --info` and `dotnet workload --info` to display the workloads version.  For example:
+
+```
+> dotnet --info
+.NET SDK:
+ Version:              8.0.201
+ Commit:               <commit>
+ Workloads version:    8.0.201
+
+<further information>
+
+> dotnet workload --info
+ Workloads version:     8.0.201
+ [wasm-tools]
+   Installation Source: SDK 8.0.100-preview.4
+   Manifest Version:    8.0.0-preview.4.23181.9/8.0.100-preview.4
+   Manifest Path:       C:\git\dotnet-sdk-main\.dotnet\sdk-manifests\8.0.100-preview.4\microsoft.net.workload.mono.toolchain.current\WorkloadManifest.json
+   Install Type:        FileBased
+```
+
+When updating workloads, console output will include the old and new workloads versions:
+
+```
+> dotnet workload install wasm-tools
+Checking for updated workloads version...
+Updating workloads version from 8.0.201-baseline.23919.2 to 8.0.201...
+Installing workload manifest microsoft.net.sdk.android version 34.0.0-preview.4.230...
+<Further workload manifest update messages>
+Installing pack Microsoft.NET.Runtime.WebAssembly.Sdk version 8.0.0-preview.4.23181.9...
+<Further workload pack installation manifests>
+Garbage collecting for SDK feature band(s) 8.0.200...
+
+Successfully updated workloads version from 8.0.201-baseline.23919.2 to 8.0.201.2.
+Successfully installed workload(s) wasm-tools.
+```
+
+The proposed [workload history](https://github.com/dotnet/sdk/pull/30486) command will also display the workloads versions.
+
+The .NET SDK will periodically (once a day, by default) check for updated workload versions in order to notify users if there is an update available.  Currently, commands such as `dotnet build` print out the following message:
+
+```
+Workload updates are available. Run `dotnet workload list` for more information.
+```
+
+The `dotnet workload list` command, in turn, prints out the following message:
+
+```
+Updates are available for the following workload(s): wasm-tools. Run `dotnet workload update` to get the latest.
+```
+
+We will modify both commands to print the same message:
+
+```
+A workload update (version 8.0.202) is available. Run `dotnet workload update` to install the latest workloads.
+```
+
+We will add a new `--version` option to `dotnet workload update` to allow installing (or downgrading to) a specific workloads version:
+
+```
+> dotnet workload update --version 8.0.202
+Updating workloads version from 8.0.201 to 8.0.202...
+<Manifest and pack installation / garbage collection messages>
+Successfully updated workloads version from 8.0.201 to 8.0.202.
+```
+
+## Specifying workload versions with global.json
+
+We will add support for specifying the workloads version in global.json.  For example:
+
+```json
+{
+  "sdk": {
+    "workloadsVersion": "8.0.201"
+  }
```

Comment:
Right now, `AutoImport.props` is imported into every project for .NET 6, 7, 8, etc. If you have 8.0.201 and 8.0.202 workloads installed, is `AutoImport.props` only imported for the pinned version?

If so, we might want to add some more robust logging to understand why the workload resolver imported one file over another.